docs/docs/spec.md

```diff
@@ -337,21 +337,31 @@ locations listed below:
 
 To determine whether a type usage appears in a null-marked scope:
 
-Look for a `@NullMarked` annotation on any of the scopes
-enclosing the type usage.
+Iterate over all the declarations that enclose the type usage, starting from the
+innermost.
 
-Class members are enclosed by classes, which may be enclosed by other class
-members or classes. and top-level classes are enclosed by packages, which may be
-enclosed by modules.
+"Enclosing" is defined as follows: Class members are enclosed by classes, which
+may be enclosed by other class members or classes. Top-level classes are
+enclosed by packages, which may be enclosed by modules.
 
 > Packages are *not* enclosed by "parent" packages.
 
-> This definition of "enclosing" likely matches
-> [the definition in the Java compiler API](https://docs.oracle.com/en/java/javase/14/docs/api/java.compiler/javax/lang/model/element/Element.html#getEnclosingElement\(\)).
+> This definition of "enclosing" largely matches
+> [the definition in the Java compiler API](https://docs.oracle.com/en/java/javase/22/docs/api/java.compiler/javax/lang/model/element/Element.html#getEnclosingElement\(\)).
+> The JSpecify definition differs slightly by skipping type-parameter
+> declarations (which cannot be annotated with declaration annotations) and by
+> defining that there exists a series of enclosing declarations for any type
```

Comment:
Really? The definition talks about class members, classes, packages, and modules. It doesn't actually say anything about parameters or local variables or cast expressions. Should it?

Reply:
...or record components or supertypes or `instanceof` or class-creation expressions or array-creation expressions or really even that, say, a return type is enclosing by its method or a type argument _inside_ the return type is enclosed by the method...

So I can't even write this one off to our attempt to ignore implementation code.

I think that you can get the right result by walking up the javac syntax tree, but that's not a great answer, even assuming that it is true today.

I've filed https://github.com/jspecify/jspecify/issues/622.